### PR TITLE
Make many more commands available as COMMAND_INT

### PR DESCRIPTION
**Pull request by peterbarker**
An alternative to https://github.com/mavlink/mavlink/pull/1982

```
Board               AP_Periph  blimp  bootloader  copter  heli  iofirmware  plane  rover  sub
Durandal                       -24    *           128     120               120    -24    -24
HerePro             24                                                                    
Hitec-Airspeed      *                                                                     
KakuteH7-bdshot                40     *           40      32                40     32     40
MatekF405                      48     *           40      48                48     48     40
Pixhawk1-1M-bdshot             72                 72      72                72     72     72
f103-QiotekPeriph   *                                                                     
f303-Universal      *                                                                     
iomcu                                                           *                         
revo-mini                      40     *           64      56                56     40     48
skyviper-v2450                                    56                                      
```

~Still need to test ROI works~

### CODE REVIEW

**Comment by magicrub** [comment]
This needs a rebase, but meanwhile I retriggered CI because this is failing DynamicRpmNotches which many other PR often fail. Feels like we need to bump this CI threshold a little.
```
Failure Summary:
  test.CopterTests2b:
    DynamicRpmNotches (Use dynamic harmonic notch to control motor noise via ESC telemetry.) (Notch-per-motor peak was higher than single-notch peak -22.193551dB > -23.800686dB) (see /tmp/buildlogs/ArduCopter-DynamicRpmNotches.txt)"
```

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 1]
```diff
@@ -4819,20 +4814,42 @@ MAV_RESULT GCS_MAVLINK::handle_command_long_packet(const mavlink_command_long_t
...
+                // can't determine a frame here as it may be above-home or above-terrain depending on vehicle.
+                return MAV_RESULT_UNSUPPORTED;
+            case MAV_CMD_DO_REPOSITION:
+                // we have never accepted this via COMMAND_LONG - don't start
+                return MAV_RESULT_UNSUPPORTED;
+            default:
+                return MAV_RESULT_UNSUPPORTED;
+            }
+        }
+        mavlink_command_int_t command_int;
```
Comment (tridge) [inline libraries/GCS_MAVLink/GCS_Common.cpp:4849]: maybe {} for init?

**Review by rmackay9 — APPROVED** [review]

**Review by tridge — CHANGES_REQUESTED** [review]
better as a table

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 2]
```diff
@@ -4819,20 +4814,42 @@ MAV_RESULT GCS_MAVLINK::handle_command_long_packet(const mavlink_command_long_t
...
 
+    case MAV_CMD_DO_SET_ROI_LOCATION:
+    case MAV_CMD_DO_SET_ROI:
+    case MAV_CMD_FIXED_MAG_CAL_YAW:
+    case MAV_CMD_COMPONENT_ARM_DISARM: {
+        // try the same thing as a COMMAND_INT:
+        MAV_FRAME frame = MAV_FRAME_GLOBAL_RELATIVE_ALT;
+        if (command_long_stores_location((MAV_CMD)packet.command)) {
+            switch (packet.command) {
+            case MAV_CMD_DO_SET_HOME:
```
Comment (tridge) [inline libraries/GCS_MAVLink/GCS_Common.cpp:4831]: missing from above list?

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 3]
```diff
@@ -4819,20 +4814,42 @@ MAV_RESULT GCS_MAVLINK::handle_command_long_packet(const mavlink_command_long_t
...
+            case MAV_CMD_DO_REPOSITION:
+                // we have never accepted this via COMMAND_LONG - don't start
+                return MAV_RESULT_UNSUPPORTED;
+            default:
+                return MAV_RESULT_UNSUPPORTED;
+            }
+        }
+        mavlink_command_int_t command_int;
+        convert_COMMAND_LONG_to_COMMAND_INT(packet, command_int, frame);
+        return handle_command_int_packet(command_int);
```
Comment (tridge) [inline libraries/GCS_MAVLink/GCS_Common.cpp:4851]: having a table would allow for much simpler code in the default: case

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/GCS_MAVLink/GCS_Common.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/GCS_MAVLink/GCS_Common.cpp]: Done in the convert method itself; this wasn't the only place which was assuming all fields came back initialised.

File: libraries/GCS_MAVLink/GCS_Common.cpp (hunk 2 above)
Comment (peterbarker) [reply libraries/GCS_MAVLink/GCS_Common.cpp]: Vestige of further work which I removed.

File: libraries/GCS_MAVLink/GCS_Common.cpp (hunk 3 above)
Comment (peterbarker) [reply libraries/GCS_MAVLink/GCS_Common.cpp]: Changed to using a table (and to just call a method here rather than in-lining the code)

**Comment by peterbarker** [comment]
> better as a table

It looks better in the code, but it does cost a few more bytes, at least with this number of commands done.

**Review by tridge — APPROVED** [review]

**Pull request merged**